python/lsst/pipe/tasks/visualizeVisit.py

```diff
@@ -0,0 +1,250 @@
+            Input exposure data to bin.
+        camera : `lsst.afw.cameraGeom.Camera`
+            Input camera to use for mosaic geometry.
+
+        Returns
+        -------
+        output : `lsst.pipe.base.Struct`
+            Results struct with attribute:
+
+            ``outputExp``
+                Binned version of input image. (`lsst.afw.image.Exposure`)
+        """
+        if inputExp.getDetector() is None:
+            detectorId = inputExp.getMetadata().get(self.config.detectorKeyword)
+            if detectorId is not None:
+                inputExp.setDetector(camera[detectorId])
+
+        binned = inputExp.getMaskedImage()
+        binned = afwMath.binImage(binned, self.config.binning)
+        outputExp = afwImage.makeExposure(binned)
+
+        if outputExp.getDetector() is None:
+            outputExp.setDetector(inputExp.getDetector())
+        return pipeBase.Struct(
+            outputExp=outputExp,
+        )
+
+
+class VisMosaicConnections(pipeBase.PipelineTaskConnections,
+                           dimensions=("instrument", )):
+    inputExps = cT.Input(
+        name="calexpBin",
+        doc="Input binned images mosaic.",
+        storageClass="ExposureF",
+        dimensions=("instrument", "detector"),
+        multiple=True,
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        doc="Input camera to use for mosaic geometry.",
+        storageClass="Camera",
+        dimensions=("instrument", "calibration_label"),
+    )
+
+    outputData = cT.Output(
+        name="calexpFP",
+        doc="Output binned mosaicked frame.",
+        storageClass="ImageF",
+        dimensions=("instrument", ),
+    )
+
+
+class VisMosaicConfig(pipeBase.PipelineTaskConfig,
+                      pipelineConnections=VisMosaicConnections):
+    """Configuration for focal plane visualization.
+    """
+    binning = pexConfig.Field(
+        dtype=int,
+        default=8,
+        doc="Binning factor to apply.",
+    )
+
+
+class VisMosaicTask(pipeBase.PipelineTask,
+                    pipeBase.CmdLineTask):
+    """Task to mosaic binned products.
+    """
+    ConfigClass = VisMosaicConfig
+    _DefaultName = 'VisMosaic'
+
+    def makeCameraImage(self, inputExps, camera, binning):
+        """Make an image of an entire focal plane.
+
+        Parameters
+        ----------
+        exposures: `dict` [`int`, `lsst.afw.image.Exposure`]
+            CCD exposures, binned by `binning`.  The keys are the
+            detectorIDs, with the values the binned image exposure.
+
+        Returns
+        -------
+        image : `lsst.afw.image.Image`
+            Image mosaicked from the individual binned images for each
+            detector.
+        """
+        class ImageSource:
+            """Source of images for makeImageFromCamera"""
+            def __init__(self, exposures):
+                self.isTrimmed = True
+                self.exposures = exposures
+                self.background = np.nan
```

Comment:
Document this with a comment, why are you doing this, why is it a class member if its only used in once method and is not alterable with init arguments. Are you expecting anyone to modify this from an instance ie foo.background? If not consider making it a local variable in getCcdImage, or if it is better as a class attribute make it private self._background